### PR TITLE
Pending Invite Screen - PHP Fatal error: Maximum execution time #909

### LINKED ISSUE #909: Pending Invite Screen - PHP Fatal error:  Maximum execution time
**Describe the bug**
When Any group have a 5k OR 7k members pending users then you will get the `PHP Fatal error:  Maximum execution time of 300 seconds exceeded`

**To Reproduce**
1. Invite 5k OR 7k members to any particular group.
2.Go to Pending Invites Screen
3. You will get white screen

**Expected behavior**
It should load the page and show the default like 20 or so...

**Screenshots**
https://prnt.sc/sh8zu0

### PR DESCRIPTION
**Pull request by shohelbuddyboss**
### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes #909.

### How to test the changes in this Pull Request:

1. Added 10724 invitation, and fixed the issues.
2. After fix, I checked and there is no more execution issue.
3.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

* Fixed Pending Invite Screen - PHP Fatal error: Maximum execution time #909

### CODE REVIEW

File: src/bp-core/classes/class-bp-invitation-manager.php [hunk 1]
```diff
@@ -378,6 +378,18 @@ public function get_invitations( $args = array() ) {
...
+		 * We are making sure that we are passing limits and page
+		 * @since 1.4.7
+		 */
+		$page = 1;
+		if (isset($_POST['page'])){
+			$page = (int) sanitize_text_field($_POST['page']);
+		}
+
+		$args['page'] = $page;
+		$args['per_page'] = 100;
```
Comment (mehulkaklotar) [inline src/bp-core/classes/class-bp-invitation-manager.php:391]: @shohelbuddyboss `per_page`  should be 20 by default.

Comment (mehulkaklotar) [reply src/bp-core/classes/class-bp-invitation-manager.php:391]: And if we are limiting the requests on the page, did we load others via pagination or load more?

Comment (shohelbuddyboss) [reply src/bp-core/classes/class-bp-invitation-manager.php]: * fixed space issue.
* per page will be load given amount of pending invitation at a time in, before it was no limit, so PHP execution time issue we saw. It's I increased to 500, so at a time, 500 pending requests will be load at a time and auto load more pagination can pull the results from 500 after 500 invitation done, users need to reload again, (Facebook also doing this)

Comment (mehulkaklotar) [reply src/bp-core/classes/class-bp-invitation-manager.php]: @shohelbuddyboss Who will tell users to reload again?

**Comment by mohit-buddyboss** [comment]
@shohelbuddyboss 
Tested this on a branch `fix-909` by creating more than 10k dummy users and sending invites between 3k to 5k. Now we are not getting an error page but another issue is appearing.

**ISSUE** When the user moves to the Pending invites tab then it shows the first list of 20 pending invites below that loader keeps moving continuously. After a few seconds, it displays a blank list of invites instead of showing a new invites list.  To understand kindly see this video: https://www.loom.com/share/48aec1ba55c74b27ad5df7334838791a

cc @mehulkaklotar @chetansatasiya

**Comment by shohelbuddyboss** [comment]
@mehulkaklotar , @chetansatasiya , @mohit-buddyboss 

This pull is no fully functional, I just tested and did fix all issue, (blank screen issue, fatal issue, load more wiped out issue). So please review this pull request and let me know if there is any more issue.

Best Regards

**Comment by chetansatasiya** [comment]
@ankitguptaindia Can you someone assign to test this?

**Comment by ankitguptaindia** [comment]
> @ankitguptaindia Can you someone assign to test this?

@chetansatasiya sure.
@mohit-buddyboss will update final status here when get chance

**Comment by mohit-buddyboss** [comment]
@shohelbuddyboss 
Tested this on a branch `fix-909` and WP version 5.5
Currently, two issues are appearing 

**Issue 1:** Tested on the group having invites between 3k to 4k. When we go to the Send invites tab it shows an Error page.
**Issue 2:** Tested by creating New Group and find out that currently, a list of only 40 members is appearing in the list. Although I have sent invites to more than 60 members. I can see those members in search results but not in the Pending invites list. 
Kindly see this video to understand the issue: https://www.loom.com/share/921c2540744241e29ca8a547137d02c0
https://prnt.sc/u2adnz

cc @ankitguptaindia @sourabhmatolia

**Comment by shohelbuddyboss** [comment]
@mohit-buddyboss 

Please check the latest changes. I've fixed those issues

cc @ankitguptaindia @sourabhmatolia

**Comment by mohit-buddyboss** [comment]
@shohelbuddyboss 
After switching to branch `fix-909`. I'm getting a fatal error on the website which not letting me access the back end as well as the front end.
Kindly see this video: https://www.loom.com/share/50fa3f3033d244f59180040dc1b218af
screenshot: https://prnt.sc/u2ukkz

@ankitguptaindia @mehulkaklotar @sourabhmatolia

**Comment by shohelbuddyboss** [comment]
@mohit-buddyboss 

Can you please check the latest changes. I've fixed the issue.

**Comment by mohit-buddyboss** [comment]
@shohelbuddyboss 
Tested this on the branch `fix-909`. 
Tested this issue in two groups. 
Group A has pending invites between 60 to 80. 
Group A has pending invites between 3k to 5k.

**Issue:** In both the groups 20 pending invites get load and after that loader keeps running continuously. I waited for more than 15 mins but the loader was kept running and not loader any more results. When I verified the ajax response it is giving this response "No pending group invitations found."  So, I think it's not finding other pending invitations. Although we can find more invitation when we search for it. 
Kindly see this video: https://www.loom.com/share/7b22e6bec7904dbfa7cc3653d5740512

cc @ankitguptaindia @sourabhmatolia

**Comment by mohit-buddyboss** [comment]
@shohelbuddyboss 

Tested it on a branch `fix-909`. Created a new group and sent an invite to more than 550 users and invites are getting a load on pending invites screen successfully and I'm not getting ajax and '+' icon error. 
Video; https://www.loom.com/share/68feee5f9fee4db9af3159a4641ae887

@chetansatasiya @mehulkaklotar @sourabhmatolia @ankitguptaindia

**Review by mohit-buddyboss — APPROVED** [review]
@shohelbuddyboss 

When I try to add new page or post then getting notice on the `fix-909` branch. These notices are not appearing on the dev branch.
Video: https://www.loom.com/share/be871ab3230843abadd92ae3cb6b05c1

@ankitguptaindia @chetansatasiya

**Comment by chetansatasiya** [comment]
@shohelbuddyboss If I have a `memory_limit` is `256` and most of the servers have a this limit and it's randomly breaking and getting a max execution time.

I debugged more and found that 

1. When a page is loading at a time it's fetching all the invited users and that is causing the issues ( Need to check why it's need a loading all the members which purpose see this https://prnt.sc/u4vid7 )
2. Once the page load after a  AJAX also it's not setting up a per page in https://prnt.sc/u4vgpj so need to fix this

**Review by chetansatasiya — DISMISSED** [review]
@shohelbuddyboss If I have a `memory_limit` is `256` and most of the servers have a this limit and it's randomly breaking and getting a max execution time.

I debugged more and found that 

1. When a page is loading at a time it's fetching all the invited users and that is causing the issues ( Need to check why it's need a loading all the members which purpose see this https://prnt.sc/u4vid7 )
2. Once the page load after a  AJAX also it's not setting up a per page in https://prnt.sc/u4vgpj so need to fix this

**Comment by shohelbuddyboss** [comment]
@chetansatasiya 

I tried to find the issue at your suggested place, seems there is no issue, I set there pagination, pagination stops the error but not working as expected.

Maybe you can make try here.

**Comment by chetansatasiya** [comment]
@shohel @shohelbuddyboss Now, I am taking over this issue.

**Review by mehulkaklotar — DISMISSED** [review]
@chetansatasiya now looks good. 👍 

needs testing here.

**Review by mehulkaklotar — APPROVED** [review]

**Comment by mohit-buddyboss** [comment]
@chetansatasiya 

Tested on the platform branch `fix-909`.
Tested by creating invites more than 600 and pending invites are loading perfectly for the newly created group and also for the old group which has more than 500 invites. Invites are getting load pagewise and on each page 20 invites are getting a load.
Video: https://www.loom.com/share/b2e9ffd9544d42fba696fd852aec2b70

**ISSUE:**
But, I've found another issue that activities all over the website are not getting a load and I'm getting 500 internal server error.
This is happening on the platform branch `fix-909` not on the dev branch. So, Kindly look into this.
**Video:** https://www.loom.com/share/16175fc7908b4242a2585834dc0a7109

@mehulkaklotar

**Comment by chetansatasiya** [comment]
> @chetansatasiya
> 
> Tested on the platform branch `fix-909`.
> Tested by creating invites more than 600 and pending invites are loading perfectly for the newly created group and also for the old group which has more than 500 invites. Invites are getting load pagewise and on each page 20 invites are getting a load.
> Video: https://www.loom.com/share/b2e9ffd9544d42fba696fd852aec2b70
> 
> **ISSUE:**
> But, I've found another issue that activities all over the website are not getting a load and I'm getting 500 internal server error.
> This is happening on the platform branch `fix-909` not on the dev branch. So, Kindly look into this.
> **Video:** https://www.loom.com/share/16175fc7908b4242a2585834dc0a7109
> 
> @mehulkaklotar

@mohit-buddyboss this is due to the `dev` branch is not mergerd in this branch. So Now I merged the `dev` branch and it looks good to me. Please test it out.

**Comment by mohit-buddyboss** [comment]
@chetansatasiya Thank you! Tested on the platform branch fix-909

FIXED > Site-wide activity not appearing issue( internal server 500 error )is resolved now and also Pending invites are getting load successfully. 
Video: https://www.loom.com/share/5f3aa9576de34f03b7be9ba89c6c7f25

@mehulkaklotar

**Review by mohit-buddyboss — APPROVED** [review]
FIXED. Related issue is resolved now and PR is ready to merge.

@mehulkaklotar

**Pull request merged**